pkg/resource/training_job/testdata/test_suite.yaml

```diff
@@ -210,4 +249,196 @@ tests:
+        desciption: "This test checks if the Controller can remove the profiler properly"
+        given:
+          desired_state: "v1alpha1/update/desired/removeProfilerRule.yaml"
+          latest_state: "v1alpha1/update/latest/profilerHappyBase.yaml"
+          svc_api: 
+            - operation: UpdateTrainingJobWithContext
+              output_fixture: "sdkapi/update/update_sucess.json"
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/update/describe/profiler_removal.json"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/removeProfilerRule.yaml"
+          error: training job is updating
+      - name: "Update=RemoveProfilerConfig"
+        desciption: "This test checks if the Controller can remove the profiler properly"
+        given:
+          desired_state: "v1alpha1/update/desired/removeProfilerConfig.yaml"
+          latest_state: "v1alpha1/update/latest/profilerHappyBase.yaml"
+          svc_api: 
+            - operation: UpdateTrainingJobWithContext
+              output_fixture: "sdkapi/update/update_sucess.json"
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/update/describe/profiler_removal.json"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/removeProfilerConfig.yaml"
+          error: training job is updating
+      - name: "Update=BothProfiler"
+        description: "Update Both Profiler parameters"
+        given:
+          desired_state: "v1alpha1/update/desired/updateBothProfileParams.yaml"
+          latest_state: "v1alpha1/update/latest/profilerHappyBase.yaml"
+          svc_api: 
+            - operation: UpdateTrainingJobWithContext
+              output_fixture: "sdkapi/update/update_sucess.json"
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/update/describe/bothprofiler_update.json"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/updateProfilerBoth.yaml"
+          error: training job is updating
+      - name: "Update=ProfilerConfig"
+        description: "Update just the profiler config"
+        given:
+          desired_state: "v1alpha1/update/desired/updateProfileConfig.yaml"
+          latest_state: "v1alpha1/update/latest/profilerHappyBase.yaml"
+          svc_api: 
+            - operation: UpdateTrainingJobWithContext
+              output_fixture: "sdkapi/update/update_sucess.json"
+            - operation: DescribeTrainingJobWithContext
+              output_fixture: "sdkapi/update/describe/profiler_config_update.json"
+        invoke: Update
+        expect:
+          latest_state: "v1alpha1/update/observed/updateProfilerConfig.yaml"
+          error: training job is updating
+      - name: "Update=ProfilerStarting"
+        description: "Tests if profiler returns a requeue error when TJ is updated in Starting."
```

Comment:
by TJ you are referring to training-job or some field?

Reply:
Training Job